app/scripts/modules/core/src/pipeline/config/triggers/pipelineRoles.component.ts

```diff
@@ -0,0 +1,46 @@
+import { IController, module } from 'angular';
+import { AuthenticationService } from 'core/authentication';
+
+class PipelineRolesController implements IController {
+  public pipeline: any;
+  public allowedRoles: string[];
+
+  public $onInit(): void {
+    this.allowedRoles = AuthenticationService.getAuthenticatedUser().roles;
+    this.pipeline.roles = this.pipeline.roles || [];
+  }
+
+  public toggleCheck(role: string) {
+    if (this.pipeline.roles.indexOf(role) === -1) {
+      this.pipeline.roles.push(role);
+    } else {
+      this.pipeline.roles.splice(this.pipeline.roles.indexOf(role), 1);
+    }
+  }
+}
+
+class PipelineRolesComponent implements ng.IComponentOptions {
+  public template = `
+  <div class="row">
+    <div class="col-md-11 col-md-offset-1">
+      <div class="row checkbox" ng-repeat="role in permissionCtrl.allowedRoles">
+        <label>
+          <input type="checkbox"
+          ng-checked="permissionCtrl.pipeline.roles.indexOf(role) > -1"
+          ng-click="permissionCtrl.toggleCheck(role)">
+          <strong>{{role}}</strong>
```

Comment:
I think this might be better as a multiselect rather than a checkbox list. If a user has many roles (maybe > 10), this list will become unwieldy. Do a search for `<ui-select>` in this repo - any with the attribute `multiple` should be a good example. The multiselect will also allow users to search for their roles.